src/main/java/org/graylog/integrations/aws/transports/KinesisTransport.java

```diff
@@ -93,37 +92,63 @@ public void handleChangedThrottledState(boolean isThrottled) {
     public void doLaunch(MessageInput input) throws MisfireException {
 
         final Region region = Region.of(Objects.requireNonNull(configuration.getString(CK_AWS_REGION)));
-        final String assumeRoleArn = configuration.getString(AWSInput.CK_ASSUME_ROLE_ARN);
         final String key = configuration.getString(CK_ACCESS_KEY);
         final String secret = configuration.getString(CK_SECRET_KEY);
-        Preconditions.checkArgument(StringUtils.isNotBlank(key), "An AWS key is required.");
-        AwsCredentialsProvider awsCredentialsProvider = AWSService.buildCredentialProvider(key, secret);
-        Preconditions.checkArgument(StringUtils.isNotBlank(secret), "An AWS secret is required.");
-
-        // Assume role ARN functionality only applies to the Kinesis runtime (not to the setup flows).
-        if (StringUtils.isNotBlank(assumeRoleArn)) {
-            StsClient stsClient = StsClient.builder()
-                                           .region(region)
-                                           .credentialsProvider(awsCredentialsProvider).build();
-            String roleSessionName = String.format("API_KEY_%s@ACCOUNT_%s",
-                                                   key,
-                                                   stsClient.getCallerIdentity().account());
-            awsCredentialsProvider = StsAssumeRoleCredentialsProvider.builder()
-                                                                     .stsClient(stsClient)
-                                                                     .refreshRequest(r -> r.roleArn(assumeRoleArn)
-                                                                                           .roleSessionName(roleSessionName)).build();
-        }
-        this.kinesisConsumer = new KinesisConsumer(
-                nodeId, this, objectMapper, kinesisCallback(input), configuration.getString(CK_KINESIS_STREAM_NAME),
-                AWSMessageType.valueOf(configuration.getString(AWSCodec.CK_AWS_MESSAGE_TYPE)), region,
-                awsCredentialsProvider,
-                configuration.getInt(CK_KINESIS_RECORD_BATCH_SIZE, DEFAULT_BATCH_SIZE)
-        );
+        final String assumeRoleArn = configuration.getString(AWSInput.CK_ASSUME_ROLE_ARN);
+
+        final String dynamodbEndpoint = configuration.getString(AWSInput.CK_DYNAMODB_ENDPOINT);
+        final String cloudwatchEndpoint = configuration.getString(AWSInput.CK_CLOUDWATCH_ENDPOINT);
+        final String iamEndpoint = configuration.getString(AWSInput.CK_IAM_ENDPOINT);
+        final String kinesisEndpoint = configuration.getString(AWSInput.CK_KINESIS_ENDPOINT);
+
+        // Validate the endpoints, because the Kinesis client will silently fail if an endpoint is invalid.
+        validateEndpoint(dynamodbEndpoint, "DynamoDB");
```

Comment:
NB: Modify validateEndpoint(Map<K,V>) to keep calling the method multiple times.

Reply:
Thanks for the suggestion. I like the idea of using a loop instead of calling a method multiple time explicitly. 

The only way I can see that working here is to create a map, call `map.put`, then iterate or use `map.forEach` as you suggested. But, the downside is that it uses the same amount of code to accomplish the same result (due to map.put). I am not sure if that will actually improve the code here. I feel like we should leave it as is. Please feel free to provide a code sample if I missed anything here.